exercises/ocr-numbers/example.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::BTreeMap;
+use std::str::Lines;
+
+fn lines_valid(lines: &Lines) -> bool {
+    lines.clone().count() % 4 == 0 && lines.clone().all(|line| line.chars().count() % 3 == 0)
+}
+
+pub fn convert(input: &str) -> Result<String, ()> {
+    if !lines_valid(&input.lines()) {
+        return Err(());
+    }
+
+    let y = input.lines().collect::<Vec<_>>();
+
+    let mut converted_lines = vec![];
+
+    for char_lines in y.chunks(4) {
+        let mut unparsed_characters = BTreeMap::new();
+
+        for line in char_lines {
+            let line_chars = line.chars().collect::<Vec<_>>();
+
+            for (char_number, char_chunk) in line_chars.chunks(3).enumerate() {
+                let char_chars = unparsed_characters.entry(char_number).or_insert(vec![]);
+                for c in char_chunk {
+                    char_chars.push(*c);
+                }
+            }
+        }
+
+        let mut parsed_characters = String::new();
+
+        for (_, v) in unparsed_characters {
```

Comment:
BTreeMap change made this much better. It's probably possible to make this a `fold`, but I don't think it's necessary.